relm4-macros/src/visitors.rs

```diff
@@ -443,6 +425,81 @@ impl VisitMut for PreAndPostView<'_> {
     }
 }
 
+/// Expands the `view_output!` macro expression in the `init` function.
+pub(crate) struct ViewOutputExpander<'errors> {
+    /// Whether a `view_output!` macro expression has been successfully expanded.
+    expanded: bool,
+
+    /// View initialization code to inject before the view output.
+    view_code: TokenStream,
+
+    /// Widgets struct initialization.
+    widgets_init: Box<syn::Expr>,
+
+    errors: &'errors mut Vec<syn::Error>,
+}
+
+impl ViewOutputExpander<'_> {
+    pub(crate) fn expand(
+        item_impl: &mut syn::ItemImpl,
+        view_code: TokenStream,
+        widgets_init: Box<syn::Expr>,
+        errors: &mut Vec<syn::Error>,
+    ) {
+        let mut expander = ViewOutputExpander {
+            expanded: false,
+            view_code,
+            widgets_init,
+            errors,
+        };
+
+        expander.visit_item_impl_mut(item_impl);
+    }
+}
+
+impl VisitMut for ViewOutputExpander<'_> {
+    fn visit_impl_item_method_mut(&mut self, method: &mut syn::ImplItemMethod) {
+        if method.sig.ident == "init" || method.sig.ident == "init_widgets" {
+            visit_mut::visit_impl_item_method_mut(self, method);
+
+            if !self.expanded {
+                self.errors.push(syn::Error::new_spanned(method, "expected an injection point for the view macro. Try using `let widgets = view_output!();`"));
+            }
+        }
+    }
+
+    fn visit_stmt_mut(&mut self, stmt: &mut syn::Stmt) {
+        let mut expand = false;
+
+        if let syn::Stmt::Local(syn::Local {
+            init: Some((_, expr)),
+            ..
+        }) = stmt
+        {
+            if let syn::Expr::Macro(mac) = &**expr {
+                if mac.mac.path.is_ident("view_output") {
+                    expand = true;
+                }
+            }
+
+            if expand {
+                mem::swap(expr, &mut self.widgets_init);
+            }
```

Comment:
What's the point of the `mem::swap` here? It looks to me like it is there to avoid uninitialized memory, but maybe you can add a comment here so it's not as confusing to read.

Reply:
Fixed.